### PR TITLE
Rubocop major version updates

### PR DESCRIPTION
**Pull request by LindseySaari**
## Description of change
**This had to be broken up into multiple PRs due to the dangerbot LOC changed limit so a few cops are disabled near the end of the .rubocop.yml config file**

As part of the gem update series, this PR updates the rubocop gem to the latest  major version. The rubocop gem introduces many new cops, all in which we're enabled via NewCops: enable in the .rubocop.yml config (new cops are considered "pending" otherwise). The --auto-gen-config flag was used to auto generate the .rubocop_todo.yml config file. Many of the cops we're corrected using the autocorrect flag. 

Some bigger updates to explicitly call out:
1. increased the Metrics/AbcSize: 44 (was 40) - this was an easy fix from the auto-generated rubocop_todo.yml that allows methods to be more complex

It's best to update each gem individually, in order to avoid issues if a roll back is needed.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#20261

## Testing Completed
- [x] CI make target passing locally 
- [x] Tested each failing (new) cop in isolation
- [x] Reviewed gem change log updates

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -349,36 +363,16 @@ RSpec/SubjectStub:
...
+Lint/NoReturnInBeginEndBlocks:
+  Exclude:
+    - 'app/services/form1010cg/service.rb'
 
-# Added in v0.84
-Lint/DeprecatedOpenSSLConstant:
-  Enabled: true
+# Will implement this in another PR
+# LOC changed is too high
+Lint/SymbolConversion:
```
Comment (LindseySaari) [inline .rubocop.yml:377]: This cop will be enabled in a separate PR. There were over 1200+ string to symbol conversions and put the LOC changed over 2500+

File: .rubocop.yml [hunk 2]
```diff
@@ -382,3 +392,30 @@ Style/SlicingWithRange:
...
+Lint/NoReturnInBeginEndBlocks:
+  Exclude:
+    - 'app/services/form1010cg/service.rb'
+
+# Will implement this in another PR
+# LOC changed is too high
+Lint/SymbolConversion:
+  Enabled: false
+
+Style/HashConversion:
```
Comment (LindseySaari) [inline .rubocop.yml:414]: HashConversion, NegatedIfElseCondition and RedundantArgument will be addressed in a separate PR as well to get below the dangerbot LOC allowed

File: modules/vaos/reporting/logs_processor.rb [hunk 3]
```diff
@@ -6,7 +6,6 @@
 require 'time'
 require 'json'
 require 'pp'
-require 'digest'
```
Comment (thilton-oddball) [inline modules/vaos/reporting/logs_processor.rb:9]: Do you know what the story is on this?

**Review by thilton-oddball — APPROVED** [review]
Just one question about a require getting removed. Otherwise looks good.

File: modules/vaos/reporting/logs_processor.rb (hunk 3 above)
Comment (LindseySaari) [reply modules/vaos/reporting/logs_processor.rb:9]: Yeah it was just a duplicate require. Already required on line 4 [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/vaos/reporting/logs_processor.rb#L4)

File: modules/vaos/app/services/vaos/middleware/vaos_logging.rb [hunk 4]
```diff
@@ -36,11 +36,7 @@ def call(env)
             log(:warn, 'VAOS service call failed!', log_tags(env, start_time, response_env))
           end
         end
-      rescue Timeout::Error, Faraday::TimeoutError => e
-        statsd_increment("#{STATSD_KEY_PREFIX}.fail", env, e)
-        log(:warn, "VAOS service call failed - #{e.message}", log_tags(env, start_time))
-        raise
-      rescue Faraday::ConnectionFailed => e
+      rescue Timeout::Error, Faraday::TimeoutError, Faraday::ConnectionFailed => e
```
Comment (LindseySaari) [inline modules/vaos/app/services/vaos/middleware/vaos_logging.rb:39]: Both rescue statements have the same code

```
statsd_increment("#{STATSD_KEY_PREFIX}.fail", env, e)
log(:warn, "VAOS service call failed - #{e.message}", log_tags(env, start_time))
raise
```

File: modules/vaos/reporting/runner.rb [hunk 5]
```diff
@@ -59,10 +59,10 @@ def usage
 # extract individual options...
 # start date
 begin
-  start_date = if !arguments[:sdt].nil?
-                 Date.parse(arguments[:sdt])
-               else
+  start_date = if arguments[:sdt].nil?
```
Comment (LindseySaari) [inline modules/vaos/reporting/runner.rb:62]: Corrected via the Style/NegatedIfElseCondition cop

File: modules/veteran/app/models/veteran/service/representative.rb [hunk 6]
```diff
@@ -19,6 +19,7 @@ class Representative < ApplicationRecord
 
       validates :poa_codes, presence: true
 
+      # rubocop:disable Lint/DuplicateBranch
```
Comment (bosawt) [inline modules/veteran/app/models/veteran/service/representative.rb:22]: Seems like the branch this is referring to actually is screwy. Might not be part of this PR, but probably worth refactoring.

**Review by bosawt — APPROVED** [review]
Looks good! Just one comment that doesn't necessarily relate to this PR!
